plugins/modules/net_tools/nmcli.py

```diff
@@ -919,6 +952,13 @@ def connection_update(self, nmcli_command):
         else:
             ifname = self.ifname
 
+        if self.type == "wifi":
+            cmd.append('ssid')
+            cmd.append(self.ssid)
+            if self.wifi_sec:
+                for name, value in self.wifi_sec.items():
+                    cmd += ['wifi-sec.%s' % name, value]
```

Comment:
a bit of a nit-picking here but
```suggestion
                    cmd.extend(['wifi-sec.%s' % name, value])
```
would be slightly more efficient.

Reply:
Thanks! The difference is actually huge:
```
[goneri@t580-gone ~]$ python3 -m timeit -s 'a = []' 'for n in range(1000): a += [n]'
5000 loops, best of 5: 82.9 usec per loop
[goneri@t580-gone ~]$ python3 -m timeit -s 'a = []' 'for n in range(1000): a.append(n)'
5000 loops, best of 5: 53.8 usec per loop
```

Reply:
Hum my bad, with the `extend()`, it's actually slow:

```
[goneri@t580-gone ~]$ python3 -m timeit -s 'a = []' 'for n in range(1000): a.extend([n])'
5000 loops, best of 5: 94.5 usec per loop
```

Reply:
It's also the case with py27:

```
[goneri@t580-gone ~]$ python2 -m timeit -s 'a = []' 'for n in range(1000): a.extend([n])'
10000 loops, best of 3: 84.6 usec per loop
[goneri@t580-gone ~]$ python2 -m timeit -s 'a = []' 'for n in range(1000): a += [n]'
10000 loops, best of 3: 70.8 usec per loop
```

Reply:
Interesting, I would have expected `extend()` to not be slower. If you add the string formatting though, the relative difference is pretty small though.

(The Python 2 performance is something I would ignore here, except if this would be something that's noticable to users - which this tiny difference clearly isn't :) )

One could argue that it's better to use `extend()` for consistency though, since it's more similar to `append()` used everywhere else. I'm not completely convinced by that. I don't care enough to ask for this to change, or ask for this to not change ;) @russoz what do you think?

Reply:
I think that for fairness we should test `extend` with lists longer than 1 element.

That said, for the sake of this PR I really don't think we should press the point. I would also prefer `append` or `extend` for consistency, but if @goneri is happy with `+` I say let's go with it.

Reply:
@russoz I tested it with two elements, `+=` is still faster.